code/game/objects/items/devices/radio/radio.dm

```diff
@@ -409,6 +409,14 @@
 /obj/item/radio/borg/Initialize(mapload)
 	. = ..()
 
+/obj/item/radio/borg/recalculateChannels()
+	var/mob/living/silicon/robot/R = loc
+	if(istype(loc))
+		for(var/channel in R.module.radio_channels)
+			channels[channel] = 1
+	. = ..()
```

Comment:
this doesn't clear channels, so if you change modules back you don't lose any channel